tests/code/testcastleuriutils.pas

```diff
@@ -133,6 +148,20 @@ procedure TTestURIUtils.TestPercentEncoding;
     But it's Ok that %4d gets converted to M, as char "M" is safe inside URI. }
   AssertEquals('file:///fooM.txt', FilenameToURISafe(URIToFilenameSafe('file:///foo%4d.txt')));
   {$endif}
+
+  {$ifdef MSWINDOWS}
+  Filename := 'C:\Users\cge\AppData\Local\test_local_filename_chars\config with Polish chars ćma źrebak żmija wąż królik.txt';
+  FilenameAsUri := FilenameToURISafe(Filename);
+  AssertEquals(FilenameAsUri, 'file:///C:/Users/cge/AppData/Local/test_local_filename_chars/config%20with%20Polish%20chars%20%C4%87ma%20%C5%BArebak%20%C5%BCmija%20w%C4%85%C5%BC%20kr%C3%B3lik.txt');
```

Comment:
A small thing: The arguments to AssertEquals are specified as "ExpectedValue, ActualValue". So the constant argument (if any) should usually be the 1st one, like

```
AssertEquals('file:///C:/Users/cge/AppData/Local/test_local_filename_chars/config%20with%20Polish%20chars%20%C4%87ma%20%C5%BArebak%20%C5%BCmija%20w%C4%85%C5%BC%20kr%C3%B3lik.txt', FilenameAsUri);
```

This only matters for clear error message in case of failure.

Reply:
Fixed :)